ios/Classes/SwiftWorkmanagerPlugin.swift

```diff
@@ -1,14 +1,135 @@
+        registrar.addMethodCallDelegate(instance, channel: foregroundMethodChannel)
+        registrar.addApplicationDelegate(instance)
+    }
+    
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        
+        switch call.method {
+        case ForegroundMethodChannel.methods.initialize.rawValue:
+            guard let callbackHandle: Int64 = call.arguments as? Int64 else {
+                result(WMPError.unexpectedMethodArguments(argumentsDescription: call.arguments.debugDescription))
+                return
+            }
+            store(callbackHandle)
+            result(nil)
+        case ForegroundMethodChannel.methods.performSampleFetch.rawValue:
+            triggerSampleBackgroundFetch()
+            result(nil)
+        default:
+            result(WMPError.unhandledMethod(methodName: call.method).asFlutterError)
+            return
+        }
+        
+    }
+    
+}
+
+//MARK: - AppDelegate conformance
+
+extension SwiftWorkmanagerPlugin {
+    
+    override public func application(_ application: UIApplication, performFetchWithCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) -> Bool {
+        
+        // First, let's retreive our callBack handle
+        guard let callbackHandle: Int64 = getStoredCallbackHandle() else {
+            completionHandler(.failed)
+            return false
+        }
+        
+        // Then, run the Flutter engine with the retreivedf callback's name and libraryPath
+        let flutterCallbackInformation: FlutterCallbackInformation = FlutterCallbackCache.lookupCallbackInformation(callbackHandle)
+        let flutterEngine = FlutterEngine.init(name: flutterThreadLabelPrefix, project: nil, allowHeadlessExecution: true)!
+        flutterEngine.run(withEntrypoint: flutterCallbackInformation.callbackName, libraryURI: flutterCallbackInformation.callbackLibraryPath)
+        
+        // Since we're now running a specific Flutter engine, no MethodChannel exists ; let's create one for WorkManager's BackgroundMethodChannel
+        let backgroundMethodChannel = FlutterMethodChannel(name: BackgroundMethodChannel.channelName, binaryMessenger: flutterEngine)
+        backgroundMethodChannel.setMethodCallHandler { (call, result) in
+            guard
+                call.method == BackgroundMethodChannel.methods.backgroundJobDidComplete.rawValue,
+                let argument: UInt = call.arguments as? UInt,
+                let backgroundFetchResult = UIBackgroundFetchResult.init(rawValue: argument)
+                else {
+                    result(WMPError.unexpectedMethodArguments(argumentsDescription: call.arguments.debugDescription))
+                    return
+            }
+            // Method name and arguments match the expected values ; call the native completionHandler
+            result(nil)
+            completionHandler(backgroundFetchResult)
+        }
+        
+        return true
```

Comment:
What does this imply?
`true` as in, the background fetch was executed successfully?  

We need to wait for the answer from Dart side and return its value here.